IPython/core/interactiveshell.py

```diff
@@ -902,42 +903,45 @@ def init_virtualenv(self):
         if 'VIRTUAL_ENV' not in os.environ:
             # Not in a virtualenv
             return
-        
-        p = os.path.normcase(sys.executable)
-        p_venv = os.path.normcase(os.environ['VIRTUAL_ENV'])
-
-        # executable path should end like /bin/python or \\scripts\\python.exe
-        p_exe_up2 = os.path.dirname(os.path.dirname(p))
-        if p_exe_up2 and os.path.exists(p_venv) and os.path.samefile(p_exe_up2, p_venv):
-            # Our exe is inside the virtualenv, don't need to do anything.
+        elif os.environ['VIRTUAL_ENV'] == '':
+            warn("Virtual env path set to '', please check if this is intended.")
             return
 
+        p = Path(sys.executable)
+        p_venv = Path(os.environ['VIRTUAL_ENV'])
+
         # fallback venv detection:
         # stdlib venv may symlink sys.executable, so we can't use realpath.
         # but others can symlink *to* the venv Python, so we can't just use sys.executable.
         # So we just check every item in the symlink tree (generally <= 3)
         paths = [p]
-        while os.path.islink(p):
-            p = os.path.normcase(os.path.join(os.path.dirname(p), os.readlink(p)))
-            paths.append(p)
-        
+        while p.is_symlink():
+            p = Path(os.readlink(p))
```

Comment:
According to [this link](https://stackoverflow.com/questions/41460434/getting-the-target-of-a-symbolic-link-with-pathlib) 
        
I believe it's not possible to replicate this on pathlib
If someone sees a better solution please let me know.

Reply:
:+1: we are 3.7+ so `p.resolve()` might work, but we can check this later.

Reply:
`p.resolve()` seems to follow symlinks all the way. Here we would like to follow them one by one to check later if any of them are inside the scope for `any(os.fspath(p_venv) in os.fspath(p) for p in paths)`